### PR TITLE
Fix: Panic when running odo list namespaces without an active Kubernetes context

### LINKED ISSUE #6316: Panic when running `odo list namespaces` without an active Kubernetes context
/kind bug

## What versions of software are you using?

**Operating System:**
Fedora 37

**Output of `odo version`:**
```
odo v3.2.0 (ce42ce435)
```

## How did you run odo exactly?
Just ran `odo list namespaces` with no active Kubernetes context:

## Actual behavior
```
$ odo list namespaces
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x28 pc=0x278ce00]

goroutine 1 [running]:
github.com/redhat-developer/odo/pkg/kclient.(*Client).GetCurrentNamespace(0x333d638?)
        /home/asoro/work/projects/odo/pkg/kclient/namespace.go:164
github.com/redhat-developer/odo/pkg/odo/genericclioptions.GenericRun({0x333e0f0, 0xc000a9cc60}, 0xc000884500, {0x46022b0, 0x0, 0x0})
        /home/asoro/work/projects/odo/pkg/odo/genericclioptions/runnable.go:184 +0xf82
github.com/redhat-developer/odo/pkg/odo/cli/list/namespace.NewCmdNamespaceList.func1(0xc000884500?, {0x46022b0?, 0x0?, 0x0?})
        /home/asoro/work/projects/odo/pkg/odo/cli/list/namespace/namespace.go:79 +0x38
github.com/spf13/cobra.(*Command).execute(0xc000884500, {0x46022b0, 0x0, 0x0})
        /home/asoro/work/projects/odo/vendor/github.com/spf13/cobra/command.go:860 +0x663
github.com/spf13/cobra.(*Command).ExecuteC(0xc00070e000)
        /home/asoro/work/projects/odo/vendor/github.com/spf13/cobra/command.go:974 +0x3b4
github.com/spf13/cobra.(*Command).Execute(...)
        /home/asoro/work/projects/odo/vendor/github.com/spf13/cobra/command.go:902
github.com/spf13/cobra.(*Command).ExecuteContext(...)
        /home/asoro/work/projects/odo/vendor/github.com/spf13/cobra/command.go:895
main.main()
        /home/asoro/work/projects/odo/cmd/odo/odo.go:70 +0x425
```

## Expected behavior
`odo list namespaces` should not panic. 

All other commands (like `odo list bindings` or `odo describe component`) just display a nicer error message, e.g.:
```
$ odo list services     
 ✗  
Please ensure you have an active kubernetes context to your cluster. 
Consult your Kubernetes distribution's documentation for more details.
Error: invalid configuration: no configuration has been provided, try setting KUBERNETES_MASTER environment variable
```

## Any logs, error output, etc?
&mdash;

### PR DESCRIPTION
**Pull request by valaparthvi**
**What type of PR is this:**
/kind bug

**What does this PR do / why we need it:**

**Which issue(s) this PR fixes:**

Fixes #6316 

**PR acceptance criteria:**

- [ ] Unit test 

- [ ] Integration test 

- [ ] Documentation 

**How to test changes / Special notes to the reviewer:**
`odo list namespace` no longer panics, instead throws an appropriate error.
```sh
KUBECONFIG=/dev/null odo list namespace
```
```sh
$ KUBECONFIG=/dev/null odo list namespace
 ✗  
Please ensure you have an active kubernetes context to your cluster. 
Consult your Kubernetes distribution's documentation for more details.
Error: invalid configuration: no configuration has been provided, try setting KUBERNETES_MASTER environment variable


```

`odo remove binding` still works when the cluster is inaccessible. The [test](https://s3.eu-de.cloud-object-storage.appdomain.cloud/odo-tests-openshift-logs/pr-6367-nocluster-tests-433.txt) for this was failing when a new REMOVE_BINDING dep was not added. Commit f542b220835b96f577975b693710cd8957914c7d fixes that.

```sh
KUBECONFIG=/dev/null odo remove binding --name something
```
```sh
$ KUBECONFIG=/dev/null odo remove binding --name something
 ✗  Service Binding "something" not found in the devfile. Available Service Bind
ings:

```

### CODE REVIEW

**Review by rm3l — CHANGES_REQUESTED** [review]
LGTM, but can you add a simple integration test case for this in `cmd_namespace_test`?

I think this should be do-able now that we can put a `nocluster` label on Ginkgo tests. Labelling the test spec with this label should make the spec run without an active Kubernetes context.

File: pkg/odo/cli/remove/binding/binding.go [hunk 1]
```diff
@@ -86,7 +86,7 @@ func NewCmdBinding(name, fullName string) *cobra.Command {
 		},
 	}
 	bindingCmd.Flags().String(backend.FLAG_NAME, "", "Name of the Binding to create")
-	clientset.Add(bindingCmd, clientset.BINDING, clientset.FILESYSTEM)
+	clientset.Add(bindingCmd, clientset.REMOVE_BINDING, clientset.FILESYSTEM)
```
Comment (feloy) [inline pkg/odo/cli/remove/binding/binding.go:89]: I'm not comfortable with this solution. If the problem is to define if KUBERNETES must be nullable or not for a specific command, we need to find a way to specify this directly

**Review by feloy — CHANGES_REQUESTED** [review]

File: pkg/odo/cli/remove/binding/binding.go (hunk 1 above)
Comment (valaparthvi) [reply pkg/odo/cli/remove/binding/binding.go:89]: Intuitively, I think KUBERNETES and KUBERNETES_NULLABLE should be mutually exclusive, but in cases where both are needed, we could only return an error if KUBERNETES_NULLABLE was not defined. WDYT?

Comment (feloy) [reply pkg/odo/cli/remove/binding/binding.go]: @valaparthvi Yes, that seems a very good idea to me

**Comment by valaparthvi** [comment]
/retest
cluster setup failure.

**Review by feloy — APPROVED** [review]

**Comment by feloy** [comment]
/override OpenShift-Integration-tests/OpenShift-Integration-tests
Flaky e2e test
```
/go/odo_1/tests/e2escenarios/e2e_test.go:19
  starting with non-empty Directory add Binding
  /go/odo_1/tests/e2escenarios/e2e_test.go:279
    [It] should verify developer workflow of using binding as env in innerloop
    /go/odo_1/tests/e2escenarios/e2e_test.go:321
```

**Pull request merged**